packages/ddp-client/client/queueStubsHelpers.js

```diff
@@ -92,9 +93,23 @@ export const loadAsyncStubHelpers = () => {
         let finished = false;
 
         Meteor._setImmediate(() => {
+          let { returnStubValue, returnServerResultPromise } = options || {};
+
           const applyAsyncPromise = oldApplyAsync.apply(this, args);
           stubPromiseResolver(applyAsyncPromise.stubPromise);
           serverPromiseResolver(applyAsyncPromise.serverPromise);
+          if (returnStubValue && returnServerResultPromise) {
+            applyAsyncPromise.stubPromise
+              .then((result) => {
+                finished = true;
+                resolve(result);
+              })
+              .catch((err) => {
+                finished = true;
+                reject(err);
+              });
+            return;
+          }
```

Comment:
With this change I break other tests like this [one](https://github.com/meteor/meteor/blob/release-3.0/packages/mongo/mongo_livedata_tests.js#L227-L269) that actually expects to handle properly any collection operation server error.

So, in Meteor 2.x, the data is created locally when running collection operations that fail on the server due to "not found" method. It feels like wrong behavior, or more difficult to mimic in the current 3.x code without fibers.

In Meteor 3.x, we can't immediately create the local data within a remote collection context knowing that we need to wait for a server promise to response and we can't just react with the stub promise straight.

Or there is something I am missing.

This issue was originally reported [here](https://meteor-community.slack.com/archives/CN350MY1G/p1704695310758029). I wonder if we can think in another way to approach this. Like change the pattern on using client collections for tests in a way that these are created as non-remote collections with `connection: null` on being instantiated on-fly and then they will effectively behave as sync collections creating the data properly on each operation.

cc @denihs